rpc/core/events.go

```diff
@@ -44,20 +44,19 @@ func Subscribe(wsCtx rpctypes.WSRPCContext, query string) (*ctypes.ResultSubscri
 
 	q, err := tmquery.New(query)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to parse a query")
-	}
-
-	err = wsCtx.AddSubscription(query, q)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to add subscription")
+		return nil, errors.Wrap(err, "failed to parse query")
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), subscribeTimeout)
 	defer cancel()
 	ch := make(chan interface{})
-	err = eventBus.Subscribe(ctx, addr, q, ch)
+	es := wsCtx.GetEventSubscriber()
+	if es == nil {
+		es = eventBus
```

Comment:
:o hmm.

Seems a bit too magical.

This whole package is a bit magical with the globals. Should we look to introduce an explicit `RPCPipe` that contains all these guys, and make all these rpc funcs methods on that ?